src/vm/readytoruninfo.cpp

```diff
@@ -148,16 +148,40 @@ BOOL ReadyToRunInfo::TryLookupTypeTokenFromName(NameHandle *pName, mdToken * pFo
             _ASSERT(TypeFromToken(tokenBasedEncloser.m_TypeToken) == mdtTypeDef);
 
             mdToken mdCurrentTypeToken = tokenBasedEncloser.m_TypeToken;
-            dwHashCode ^= GetVersionResilientTypeHashCode(tokenBasedEncloser.m_pModule->GetMDImport(), mdCurrentTypeToken);
+
+            bool hasError = false;
+            EX_TRY
+            {
+                dwHashCode ^= GetVersionResilientTypeHashCode(tokenBasedEncloser.m_pModule->GetMDImport(), mdCurrentTypeToken);
```

Comment:
Why is it no ok to throw from here? This method has the special `FORBIDGC_LOADER_USE_ENABLED()` contract that says it is ok to throw from here, except for the case where we are in the GC stackwalk. But by the time we got here for GC stackwalk, we should have taken this path before - and thus we can assume that everything will be good.